MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -321,37 +321,49 @@ class FixedLengthQueue<T> {
     }
 }
 
-class CoreFeedbackEvent: Hashable {
+public class CoreFeedbackEvent: Hashable {
     var id = UUID()
     
     var timestamp: Date
     
-    var eventDictionary: [String: Any]
+    public var eventDictionary: [String: Any]
     
     init(timestamp: Date, eventDictionary: [String: Any]) {
         self.timestamp = timestamp
         self.eventDictionary = eventDictionary
     }
     
-    var hashValue: Int {
+    public var hashValue: Int {
         get {
             return id.hashValue
         }
     }
     
-    static func ==(lhs: CoreFeedbackEvent, rhs: CoreFeedbackEvent) -> Bool {
+    public static func ==(lhs: CoreFeedbackEvent, rhs: CoreFeedbackEvent) -> Bool {
         return lhs.id == rhs.id
     }
 }
 
-class FeedbackEvent: CoreFeedbackEvent {
-    func update(type: FeedbackType, description: String?) {
-        eventDictionary["feedbackType"] = type.description
-        eventDictionary["description"] = description
+public class FeedbackEvent: CoreFeedbackEvent {
+    public var type: FeedbackType? {
+        didSet {
+            self.eventDictionary["feedbackType"] = type?.rawValue
+        }
+    }
+    
+    public var description: String? {
+        didSet {
+            eventDictionary["description"] = description
+        }
+    }
+    
+    public var coordinate: CLLocationCoordinate2D? {
+        guard let latitude = eventDictionary["lat"] as? CLLocationDegrees, let longitude = eventDictionary["lng"] as? CLLocationDegrees else { return nil }
```

Comment:
This was a quick-and-dirty way for me to get at the underlying values in the model. My suggestion for production would be that we represent this data in our model class (or protocol) with concrete properties, and synthesize their inclusion into a dictionary model.